src/ModelContextProtocol.Core/Server/AIFunctionMcpServerTool.cs

```diff
@@ -185,6 +186,9 @@ private static McpServerToolCreateOptions DeriveOptions(MethodInfo method, McpSe
         {
             newOptions.Description ??= descAttr.Description;
         }
+        
+        var filters = method.GetCustomAttributes<ToolFilter>().OrderBy(f => f.Order).ToArray<IToolFilter>();
```

Comment:
This won't be compatible with aot compilation. That is likely a blocker.

Reply:
`GetCustomAttribute<>` is already used (5 lines above my code 🙂).
That's how the tools metadata and description are extracted from `[McpServerTool]` and `[Description]`.